src/artifacts.js

```diff
@@ -509,6 +509,16 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
       key: artifact.details.key,
     };
 
+    res.set('x-amz-meta-runid', runId);
+    res.set('x-amz-meta-taskid', taskId);
+    res.set('x-amz-meta-name', name);
+    res.set('x-amz-meta-content-sha256', artifact.details.contentSha256);
+    res.set('x-amz-meta-content-length', artifact.details.contentLength);
+    res.set('x-amz-meta-transfer-sha256', artifact.details.transferSha256);
+    res.set('x-amz-meta-transfer-length', artifact.details.transferLength);
```

Comment:
Can we have please have a separate discussion about this...
I'm not sure it's a good idea to do the `x-amz-meta-` headers in our code...

I'm even less sure that it is wise to store things like `transfer-sha256`.
And I see `runid` and `taskid` as useless here...

If you want to add a `x-taskcluster-redirect-content-sha256` (or similar name) I would fully support that. But this seems like it's lacking separation of concerns, coupling too many things too tightly. Are we even storing these headers; I really don't want to do that...

This seems like we coupling the implemention of server and client too tightly, we won't be able to migrated from S3, should we ever need to use a different blob store.

Reply:
runid, taskid and name being stored on the actual backing artifact is *really* nice because it allows us to easily rebuild the data about the artifacts stored, and because it is ensured to be correct by remotely-signed-s3 request signing, we can use it to repopulate the datastore, should we lose it.

As well, it allows someone who gets an opaque url to programatically know the name, taskid and runid of the resulting artifact without having needed to speak to the queue.  This would be useful doing debugging around integration of artifacts into the build system possibly.

I don't care so much about the name, but I will say that the overhead to change x-amz-meta to x-azure-meta on an informational header on a redirect would be roughly 0.00000001% of the effort of supporting a second artifact backend.  If we do rename it, let's go with "X-Taskcluster-%s" and drop the unnecessary '-redirect' token.

This is not tight coupling.  These are informational headers, not something stored.  It's totally possible to use a field called 'x-amz-meta-content-sh256' in an azure system.  It's a name, and we can add/change it as needed.

Reply:
> we can use it to repopulate the datastore, should we lose it.

That's a solid argument... I'm also okay with store the metadata.

But why do we want to expose it during the redirect? That seems overkill.
Except for exposing the hash of what we're redirecting to, that seems nice.

Okay, I'll try to not mind informational headers :) hehe

Reply:
so this is one of those things that I want for easier debugging.  I know it's wasted bytes and I know it's not needed, but things like this can make debugging a lot easier.

Reply:
sounds like we're in agreement.  I'll do some renames and submit a new PR